src/daemon/rpmostreed-os.c

```diff
@@ -735,8 +749,15 @@ os_merge_or_start_deployment_txn (RPMOSTreeOS            *interface,
                                           fd_list,
                                           &local_error);
       if (transaction)
-        rpmostreed_transaction_monitor_add (self->transaction_monitor,
-                                            transaction);
+        rpmostreed_transaction_monitor_add (self->transaction_monitor, transaction);
+
+      /* For the AutomaticUpdateTrigger "check" and "download" cases, we want to make sure
+       * we refresh CachedUpdate after; "deploy" will do this through sysroot_changed */
+      const char *method_name = g_dbus_method_invocation_get_method_name (invocation);
+      if (g_str_equal (method_name, "AutomaticUpdateTrigger") &&
+          (default_flags & (RPMOSTREE_TRANSACTION_DEPLOY_FLAG_DOWNLOAD_ONLY |
+                            RPMOSTREE_TRANSACTION_DEPLOY_FLAG_DOWNLOAD_METADATA_ONLY)))
+        g_signal_connect (transaction, "closed", G_CALLBACK (on_auto_update_done), self);
```

Comment:
Move this to `os_handle_automatic_update_trigger()` after the call to `os_merge_or_start_deployment_txn()`?  (Also holy cow do we have a lot of parameters there now)

Reply:
But that will run just after we start the transaction is started, right? We want to update the parameters *after* the transaction is done.